WordPress/gutenberg
# LineHeightControl: Allow for more granular control of decimal places

## What?

This adds an additional `step` value to the `LineHeightControl` of `0.01`, which allows for more granular control over the line-height value.

Currently, users are only able to use values with single decimal places, such as 1.5 or 1.6. But there is quite a big difference between just one decimal place, and it would be great if smaller values could be set, such as 1.55, etc.

Please see https://github.com/WordPress/gutenberg/issues/42998 and https://github.com/WordPress/gutenberg/issues/51754 for more details.

## Why?

Closes https://github.com/WordPress/gutenberg/issues/42998.

## How?

Adds a `spinFactor` prop to the NumberControl component to opt-in for more granular control over decimal places.

## Testing Instructions




1. Go to the Post or Site editor.
2. Insert a Paragraph block or any block that supports line-height.
3. Try changing the line-height using the block controls to a value with two decimal places, e.g. 1.62.
4. The value should not automatically round up or down to a value with one decimal place and should remain the same value with two decimal places.
5. Using the arrow keys or the spin controls, increase and decrease the line-height. The value should change by 0.1.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1645628/b6f4c78e-9861-4461-bc2c-94ab5ed834eb

## Dev note

A new `spinFactor` prop has been added to `NumberControl`, allowing consumers of the components to specify by how much should the input's value increment/decrement with respect to the `step` prop. This is particularly useful when more granular control is needed (thus allowing for more decimal places and a smaller `step`) but without sacrificing the UX around manually incrementing/decrementing the value.

Even if the prop was added to `NumberControl`, all components based on `NumberControl` can benefit from this change — this includes, for example, `UnitControl` and `LineHeightControl`.
